### ISSUE TITLE
keywords missing via OAI

### ISSUE THREAD

**Issue by bozana** [pkp/pkp-lib#4242]
the additional metadata like keywords and discipline are wrongly addressed/accessed in the code to provide them in oai_dc, s. https://github.com/pkp/ojs/blob/ojs-stable-3_1_1/plugins/metadata/dc11/filter/Dc11SchemaArticleAdapter.inc.php#L93-L94.
It should be used something like: https://github.com/pkp/ojs/blob/ojs-stable-3_1_1/plugins/generic/googleScholar/GoogleScholarPlugin.inc.php#L102-L103.

**Comment by bozana** [comment]
PRs:
ojs-stable-3_1_1: https://github.com/pkp/ojs/pull/2169

**Comment by bozana** [comment]
@asmecher, could you take a look at this PR?

**Comment by asmecher** [comment]
That looks good -- please continue, thanks!

**Comment by bozana** [comment]
sorry @asmecher, I've just seen that OMP is displaying each DC subject in its own dc:subject element and it does not consider discipline. Shall we do it in the same way for OJS 3.1?

**Comment by asmecher** [comment]
We should definitely standardize between the two apps, but I'd suggest asking Mike Nason (e.g. via Slack) for his recommendation. He's looking into OJS metadata improvements.

**Comment by AhemNason** [comment]
Hey! So, from what I can tell, "discipline" isn't something covered by a dc:subject field. 

```
dcterms:subject
    dcterms:description "Typically, the subject will be represented using keywords, key phrases, or classification codes. Recommended best practice is to use a controlled vocabulary."@en ;
    dcterms:hasVersion <http://dublincore.org/usage/terms/history/#subjectT-002> ;
    dcterms:issued "2008-01-14"^^<http://www.w3.org/2001/XMLSchema#date> ;
    dcterms:modified "2012-06-14"^^<http://www.w3.org/2001/XMLSchema#date> ;
    a rdf:Property ;
    rdfs:comment "The topic of the resource."@en ;
    rdfs:isDefinedBy <http://purl.org/dc/terms/> ;
    rdfs:label "Subject"@en ;
    rdfs:subPropertyOf <http://purl.org/dc/elements/1.1/subject> ;
    skos:note "This term is intended to be used with non-literal values as defined in the DCMI Abstract Model (http://dublincore.org/documents/abstract-model/).  As of December 2007, the DCMI Usage Board is seeking a way to express this intention with a formal range declaration."@en .
```

I've pasted the dc:terms subject field here for more specificity. It's pretty clear that the intent of a subject field is mostly just keyword/keyphrase with, ideally, some sort of controlled vocabulary. Making a controlled vocabulary in this space would be a nightmare. Generally, a basic dc record wouldn't contain much more than the following: http://dublincore.org/documents/dces/ 

There _is_ a way to relay this information very specifically but it's likely more onerous than most publishers would consider. That's via subject headings. Under section 4 of the [dces documentation](http://dublincore.org/documents/dcmi-terms/) there are elements for LCSH (Library of Congress Subject Headings) and MESH. There's an NLM option as well. But these are fields that are very unlikely to be filled out by anyone other than cataloguers. Most users would never use them or, wouldn't know how to use them properly. 

I can see from here that the library of congress is just mapping these to subject as well but with an attribute? https://www.loc.gov/marc/marc2dc.html I'd consider MARC to be pretty authoritative here.

TL;DR, if we're pushing out via OAI, subjects should be restricted to keywords. If we want to preserve a discipline field, that should be local metadata only. There's no place in dublin core to which we could appropriately map that.

**Comment by bozana** [comment]
Thanks @AhemNason! So OMP seems to do it correct... without discipline...
Maybe then just one more question: shall we provide each keyword in a separate dc:subject element? 
From DRIVER Guidelines: 

> For keywords/keyphrases that are not controlled by a vocabulary or
> thesaurus either encode multiple terms with a semi-colon separating
> each keyword/keyphrase; or repeat the element for each term. There
> are no requirements regarding the capitalization of keywords though
> internal (within archive) consistency is recommended.
> Where terms are taken from a standard classification schema: encode
> each term in a separate element. Encode the complete subject
> descriptor according to the relevant scheme. Use the capitalisation and
> punctuation used in the original scheme.

It seems to me that we are on the safe side that way... Also, because the keywords are now stored separately (and not as a string with ';'), it would be easier to do so...
Thanks a lot!

**Comment by AhemNason** [comment]
Yeah, that's the safe side for sure. I don't anticipate many people use an actual classification of any kind in this space but I agree that multliple elements is the way to go.
